components/Tables/ContributorsTable.tsx

```diff
@@ -0,0 +1,340 @@
+                </button>
+              )
+            );
+          },
+        },
+      ],
+    },
+  ];
+
+  const table = useReactTable({
+    columns: useMemo(() => (isMobile ? mobileColumns : defaultColumns), [isMobile]),
+    data: contributors ?? [],
+    manualSorting: true,
+    manualPagination: true,
+    getCoreRowModel: getCoreRowModel(),
+    getRowCanExpand: () => isMobile,
+    onRowSelectionChange: setSelectedContributors,
+    getExpandedRowModel: getExpandedRowModel(),
+    getRowId: (row) => row.login,
+    state: { sorting, rowSelection: selectedContributors },
+  });
+
+  return (
+    <div className="flex flex-col gap-4">
+      {Object.keys(selectedContributors).length > 0 && (
+        <div className="flex justify-between">
+          <p>{Object.keys(selectedContributors).length} selected</p>
+          {isMobile ? (
+            <AddToContributorInsightDrawer repository={repository} contributors={Object.keys(selectedContributors)} />
+          ) : (
+            <Button variant="primary" onClick={() => setIsAddToContributorInsightModalOpen(true)}>
+              Add to Insight
+            </Button>
+          )}
+        </div>
+      )}
+
+      <Table>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <TableHead key={header.id} className="font-semibold">
+                  {header.column.columnDef.enableSorting ? (
+                    <button
+                      onClick={() => {
+                        const { enableSorting } = header.column.columnDef;
+                        const isAscending = Boolean(sorting.find((item) => item.id === header.id && !item.desc));
+
+                        if (enableSorting) {
+                          setOscrSorting(isAscending ? "ASC" : "DESC");
+
+                          setSorting((currentState) => {
+                            // future-proof, set other column sorting to false
+                            const state = currentState
+                              .filter((item) => item.id !== header.id)
+                              .map(({ id }) => ({ id, desc: false }));
+
+                            return [
+                              ...state,
+                              {
+                                id: header.id,
+                                desc: isAscending,
+                              },
+                            ];
+                          });
+                        }
+                      }}
+                      className="flex gap-2 w-fit items-center"
+                    >
+                      <h2 className="font-semibold">{header.column.columnDef.header?.toString()}</h2>
+                      {header.column.columnDef.enableSorting &&
+                        (Boolean(sorting.find((item) => item.id === header.id && !item.desc)) ? (
```

Comment:
`header.column.getIsSorted()` is a dedicated TanStack API for this logic. Returns `"asc" | "desc" | false`